python/pyarrow/parquet/core.py

```diff
@@ -2782,9 +2782,9 @@ def partitioning(self):
 
 
 def read_table(source, *, columns=None, use_threads=True, metadata=None,
-               schema=None, use_pandas_metadata=False, memory_map=False,
-               read_dictionary=None, filesystem=None, filters=None,
-               buffer_size=0, partitioning="hive", use_legacy_dataset=False,
+               schema=None, use_pandas_metadata=False, read_dictionary=None,
+               memory_map=False, buffer_size=0, partitioning="hive",
+               filesystem=None, filters=None, use_legacy_dataset=False,
```

Comment:
This seems breaking to me, though I'm not sure if it's of much concern. Feedback appreciated.

Reply:
This shouldn't be a deprecating change, since the parameters here are keyword-argument-only (only the first `source` argument is allowed as positional argument, due to the `*` in the signature)